taurus_pyqtgraph/datainspectormodel.py

```diff
@@ -0,0 +1,205 @@
+from datetime import datetime
+
+from pyqtgraph import TextItem, InfiniteLine
+import numpy as np
+from taurustrendset import TaurusTrendSet
+
+
+class DataInspectorModel(object):
+    """
+    DataInspectorModel is a general class use to point-picking.
+    This class use a numpy library to finding the nearest point
+    from the :class:`pyqtgraph.InfiniteLine`.
+    Attach two widget to the plot:
+        - :class:`pyqtgraph.InfiniteLine` - to selecting point
+        - :class:`pyqtgraph.TextItem` - to showing the x,y value
+    Also this class blink the selected scatter point item.
+    """
+    def __init__(self, plot):
+
+        self.plot = plot
+
+        self.scatter = {}
+        self.informationClouds = {}
+        # Maybe good idea is give user's access
+        # to they have possibility to change the cloud style?
+        self.cloudStyle = "background-color: #35393C;"
+        self.custom_point_style_init = False
+
+        self.date_format = "%Y-%m-%d %H:%M:%S"
+
+        self.v_line = InfiniteLine(angle=90, movable=False)
+        self.plot.addItem(self.v_line, ignoreBounds=True)
+
+    def __del__(self):
+        """
+        Destructor which removing all data inspector item's
+        """
+        self.detach()
+
+    def mouseMoved(self, evt):
+        """
+        Slot use to handle the mouse move event, and preform the action
+        on the plot.
+
+        :param evt: mouse event
+        """
+        pos = evt[0]
+
+        if self.plot.sceneBoundingRect().contains(pos):
+            mousePoint = self.plot.vb.mapSceneToView(pos)
+
+            # Move InfiniteLine to the new position
+            self.v_line.setPos(mousePoint.x())
+
+            # Iteration over each curves in the plot and checking
+            # that mouse point if is near the x axis date
+            for i, curve in enumerate(filter(lambda o: not isinstance(o, TaurusTrendSet),
```

Comment:
why filtering out curves that are not trendsets?. IMHO this tool should work for any curve being displayed in the plot item.
Note that the idea is that the tools that we provide should be as generic (and taurus-independent) as possible.